Add mint score command to CLI reference (#5280)
mintlify/docs: cli/index.mdx

### Run quality checks
Run `mint broken-links` to find broken links, `mint a11y` to check accessibility, and `mint validate` to validate your documentation build before it reaches production.

### Check agent readiness

> Run `mint score --url <url>` to evaluate how well a documentation site is prepared for AI agent consumption. The command displays an overall readiness score and individual check results.

Reviewer: ```suggestion
Run `mint score <url>` to evaluate how well agents can navigate a documentation site. The command displays an overall readiness score and individual check results.
```
